clientv3/client.go

```diff
@@ -280,9 +283,55 @@ func newClient(cfg *Config) (*Client, error) {
 		logger.Set(log.New(ioutil.Discard, "", 0))
 	}
 
+	go client.autoSync()
 	return client, nil
 }
 
+// UpdateEndpoints updates clients endpoints.
+func (c *Client) UpdateEndpoints(eps ...string) {
+	epm := make(map[string]struct{})
+	for _, ep := range eps {
+		epm[ep] = struct{}{}
+	}
+	c.balancer.update(epm)
+}
+
+// Sync syncs client's endpoints in case membership change happens.
+func (c *Client) Sync() error {
+	mresp, err := c.MemberList(c.ctx)
+	if err != nil {
+		return err
+	}
+	epm := make(map[string]struct{})
+	for _, m := range mresp.Members {
+		for _, ep := range m.ClientURLs {
+			epm[ep] = struct{}{}
+		}
+	}
+	if len(c.lastAllEps) == 0 {
+		c.lastAllEps = epm
+	}
+	if !reflect.DeepEqual(c.lastAllEps, epm) {
+		// membership has changed
+		c.balancer.update(epm)
+	}
+	return nil
+}
+
+func (c *Client) autoSync() {
+	if c.cfg.AutoSync == time.Duration(0) {
```

Comment:
this check should probably decide whether to launch the goroutine instead of happening after the goroutine starts